source/Preferences.cpp

```diff
@@ -112,15 +112,12 @@ void Preferences::Load()
 			screenModeIndex = max<int>(0, min<int>(node.Value(1), SCREEN_MODE_SETTINGS.size() - 1));
 		else if(node.Token(0) == "alert indicator")
 			alertIndicatorIndex = max<int>(0, min<int>(node.Value(1), ALERT_INDICATOR_SETTING.size() - 1));
-		else if(node.Token(0) == "previous saves" && node.Size() >= 2)
-			previousSaveCount = node.Value(1);
+		else if(node.Token(0) == "previous saves")
+			previousSaveCount = max<int>(3, node.Value(1));
```

Comment:
You should still check that there's a value at 1 before accessing it.
```suggestion
			previousSaveCount = max<int>(3, node.Size() >= 2 ? node.Value(1) : 3);
```

Reply:
`DataNode::Value()` checks and returns `0.` if there isn't.

Reply:
It also logs an error.

Reply:
If there's a line `"previous saves"` with no second token in the preferences file, something has gone wrong, so it probably should.
~~Though, not all the other entries here are consistent on this.~~